shared/config/config.go

```diff
@@ -30,10 +30,15 @@ type Pre2PConfig struct {
 }
 
 type P2PConfig struct {
-	Protocol   string          `json:"protocol"`
-	Address    pcrypto.Address `json:"address"`
-	ExternalIp string          `json:"external_ip"`
-	Peers      []string        `json:"peers"`
+	Protocol         string          `json:"protocol"`
+	Address          pcrypto.Address `json:"address"`
```

Comment:
It should, but since this PR is taken from the original P2P Module, keeping it as pcrypto.Address silenced errors I was not willing to deal with at that particular moment. Let's keep as is I will fix regardless. No need for a TODO as it already exists in the parts of the code that I will come in the next PRs.